agate/agate/tests/test_api_endpoints.py

```diff
@@ -0,0 +1,82 @@
+from rest_framework import status
+from rest_framework.test import APITestCase
+from agate.models import IngestionAttempt
+from agate.caching import TokenCache
+from django.urls import reverse
+import hashlib
+
+auth = "my_token"
+
+
+class IngestionAttemptAPITests(APITestCase):
+
+    def setUp(self):
+        # Setup: Create some IngestionAttempt instances for testing
+        IngestionAttempt.objects.create(uuid="user1",
+                                        is_published=True,
+                                        project="project",
+                                        site="here",
+                                        is_test_attempt=False)
+        IngestionAttempt.objects.create(uuid="user2",
+                                        is_published=True,
+                                        project="project",
+                                        site="here",
+                                        is_test_attempt=False)
+
+        TokenCache.objects.create(
+            projects_output='{"data": [{"project":"project"}]}',
+            site_output="here",
+            token_hash=hashlib.sha256(auth.encode("utf-8")).hexdigest()
+        )
+
+    def test_unauthorized_access(self):
+        # Test if the API correctly denies unauthorized access
+        response = self.client.get(reverse("agate:ingestion"))
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_get_ingestion_attempts(self):
+        # Test if GET returns the list of ingestion attempts
+        response = self.client.get(reverse("agate:ingestion"), data={"project": "project"}, HTTP_Authorization=auth)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data['results']), 2)  # Expecting 2 ingestion attempts
+        self.assertEqual(response.data['previous'], None)
+        self.assertEqual(response.data['next'], None)
```

Comment:
I don't think it matters in terms of behaviour but the code and error messages are a bit clearer if we use some of the specific [`unittest` assert methods](https://docs.python.org/3/library/unittest.html#assert-methods), e.g.
```suggestion
        self.assertIsNone(response.data['previous'])
        self.assertIsNone(response.data['next'])
```